packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts

```diff
@@ -254,6 +256,36 @@ export class MetricsExplorerService<
         );
     }
 
+    private async _getFirstNSegments(
+        user: SessionUser,
+        projectUuid: string,
+        exploreName: string,
+        segmentDimension: string | null,
+        metricQuery: MetricQuery,
+    ) {
+        if (!segmentDimension) {
+            return [];
+        }
+
+        const getSegmentsMetricQuery: MetricQuery = {
+            ...metricQuery,
+            exploreName,
+            dimensions: [segmentDimension],
+            sorts: [{ fieldId: segmentDimension, descending: true }],
```

Comment:
I believe the sort here should be by a metric

Reply:
Yep, that's true, changed in most recent commit! Thank you 🙏